### PR TITLE
Clop ransomware

### PR DESCRIPTION
**Pull request by tccontre**
### Analytics Story:
**Clop Ransomware**

**### **New Detections:****
clop_common_exec_parameter.yml
clop_ransomware_known_service_name.yml
create_service_in_suspicious_file_path.yml
high_file_deletion_frequency.yml
high_process_termination_frequency.yml
process_deleting_its_process_file_path.yml
ransomware_notes_bulk_creation.yml
resize_shadowstorage_volume.yml

### New Test File
clop_common_exec_parameter.test.yml
clop_ransomware_known_service_name.test.yml
create_service_in_suspicious_file_path.test.yml
high_file_deletion_frequency.test.yml
high_process_termination_frequency.test.yml
process_deleting_its_process_file_path.test.yml
ransomware_notes_bulk_creation.test.yml
resize_shadowstorage_volume.test.yml

### CODE REVIEW

**Comment by P4T12ICK** [comment]
This PR was created by Automated Detection Testing Service :robot:

Tested detections:

Clop Common Exec Parameter :white_check_mark:
- diskUsage: 401408
- runDuration: 52.805s
- scanCount: 86562

**Comment by P4T12ICK** [comment]
This PR was created by Automated Detection Testing Service :robot:

Tested detections:

Clop Ransomware Known Service Name :x:
- diskUsage: 61440
- runDuration: 0.003s
- scanCount: 0

**Comment by P4T12ICK** [comment]
This PR was created by Automated Detection Testing Service :robot:

Tested detections:

High Process Termination Frequency :x:
- diskUsage: 61440
- runDuration: 0.003s
- scanCount: 0

**Comment by P4T12ICK** [comment]
This PR was created by Automated Detection Testing Service :robot:

Tested detections:

High File Deletion Frequency :x:
- diskUsage: 65536
- runDuration: 0.003s
- scanCount: 0

**Comment by P4T12ICK** [comment]
This PR was created by Automated Detection Testing Service :robot:

Tested detections:

Process Deleting Its Process File Path :x:
- diskUsage: 65536
- runDuration: 0.003s
- scanCount: 0

**Comment by P4T12ICK** [comment]
This PR was created by Automated Detection Testing Service :robot:

Tested detections:

Ransomware Notes bulk creation :white_check_mark:
- diskUsage: 188416
- runDuration: 1.112s
- scanCount: 436

**Comment by P4T12ICK** [comment]
This PR was created by Automated Detection Testing Service :robot:

Tested detections:

Process Deleting Its Process File Path :x:
- diskUsage: 65536
- runDuration: 0.003s
- scanCount: 0

**Comment by P4T12ICK** [comment]
This PR was created by Automated Detection Testing Service :robot:

Tested detections:

High Process Termination Frequency :x:
- diskUsage: 61440
- runDuration: 0.003s
- scanCount: 0

**Comment by P4T12ICK** [comment]
This PR was created by Automated Detection Testing Service :robot:

Tested detections:

Resize ShadowStorage volume :white_check_mark:
- diskUsage: 524288
- runDuration: 1.958s
- scanCount: 16

**Comment by P4T12ICK** [comment]
This PR was created by Automated Detection Testing Service :robot:

Tested detections:

Create Service In Suspicious File Path :white_check_mark:
- diskUsage: 135168
- runDuration: 0.749s
- scanCount: 21

**Comment by P4T12ICK** [comment]
This PR was created by Automated Detection Testing Service :robot:

Tested detections:

High File Deletion Frequency :x:
- diskUsage: 65536
- runDuration: 0.003s
- scanCount: 0

**Comment by trogdorsey** [comment]
Endpoint detections should output and group by process ids not by process names

**Comment by tccontre** [comment]
thanks David for this great tips and advice. will take note of that and add it to the detections rules. :)

**Comment by P4T12ICK** [comment]
This PR was created by Automated Detection Testing Service :robot:

Tested detections:

Clop Ransomware Known Service Name :white_check_mark:
- diskUsage: 131072
- runDuration: 0.739s
- scanCount: 21

**Comment by P4T12ICK** [comment]
This PR was created by Automated Detection Testing Service :robot:

Tested detections:

Process Deleting Its Process File Path :white_check_mark:
- diskUsage: 143360
- runDuration: 0.9510000000000001s
- scanCount: 145

**Comment by P4T12ICK** [comment]
This PR was created by Automated Detection Testing Service :robot:

Tested detections:

High File Deletion Frequency :white_check_mark:
- diskUsage: 147456
- runDuration: 1.108s
- scanCount: 485

**Comment by P4T12ICK** [comment]
This PR was created by Automated Detection Testing Service :robot:

Tested detections:

High Process Termination Frequency :white_check_mark:
- diskUsage: 139264
- runDuration: 26.816s
- scanCount: 46708

**Review by P4T12ICK — DISMISSED** [review]
Looks good to me

File: detections/endpoint/high_process_termination_frequency.yml [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+  encrypting files. This technique is designed to avoid an exception error while accessing
+  (docs, images, database and etc..) in the infected machine for encryption.
+search: '`sysmon` EventCode=5 |bin _time span=3s |stats values(Image) as proc_terminated
+  min(_time) as firstTime max(_time) as lastTime  count by Computer EventCode ProcessID
+  | where count >= 15 | `security_content_ctime(firstTime)`| `security_content_ctime(lastTime)`
+  | `high_process_termination_frequency_filter`'
+how_to_implement: To successfully implement this search, you need to be ingesting
+  logs with the Image (process full path of terminated process) from your endpoints.
+  If you are using Sysmon, you must have at least version 6.0.4 of the Sysmon TA.
+known_false_positives: UPDATE_KNOWN_FALSE_POSITIVES
```
Comment (josehelps) [inline detections/endpoint/high_process_termination_frequency.yml]: needs to be filled

**Review by P4T12ICK — APPROVED** [review]

**Pull request merged**